kale/worker.py

```diff
@@ -360,25 +360,30 @@ def _run_single_message(self, message, time_remaining_sec):
         :param time_remaining_sec: int seconds left for the batch of messages. Used mainly for logs
         """
         task_inst = message.task_inst
+
         try:
             with timeout.time_limit(task_inst.time_limit):
-                self.run_task(message)
-        except Exception as err:
-            # Re-publish failed tasks.
-            # As an optimization we could run all of the failures from a
-            # batch together.
-            permanent_failure = not task_inst.__class__.handle_failure(
-                message, err)
-            if permanent_failure and settings.USE_DEAD_LETTER_QUEUE:
-                self._permanent_failures.append(message)
-
-            self._failed_messages.append(message)
-
-            self._on_task_failed(message, time_remaining_sec, err,
-                                 permanent_failure)
-        else:
-            self._successful_messages.append(message)
-            self._on_task_succeeded(message, time_remaining_sec)
+                if self.should_run_task(message):
```

Comment:
rearrange this so that if we should not run the task, we republish and then return instead of indenting everything further